--- a/Source/WebKit/gtk/webkit/webkitwebview.cpp
+++ b/Source/WebKit/gtk/webkit/webkitwebview.cpp
@@ -804,6 +804,12 @@
     if (!frame.view())
         return FALSE;
 
+#ifndef GTK_API_VERSION_2
+    /* Ignore Alt-scroll events in GTK3 so that GtkNotebook can use it to switch tabs */
+    if (event->state & GDK_MOD1_MASK)
+        return FALSE;
+#endif
+
     PlatformWheelEvent wheelEvent(event);
     return frame.eventHandler().handleWheelEvent(wheelEvent);
 }
