From: Adrian Perez de Castro <aperez@igalia.com>
Subject: Allow toggling ENABLE_WPE_PLATFORM
Bug: https://bugs.webkit.org/show_bug.cgi?id=279879
Origin: https://github.com/WebKit/WebKit/commit/eabcab4b0d8ab44755605788848c4f79d818d5c8
Index: webkitgtk/Source/WebKit/UIProcess/API/glib/WebKitWebView.h.in
===================================================================
--- webkitgtk.orig/Source/WebKit/UIProcess/API/glib/WebKitWebView.h.in
+++ webkitgtk/Source/WebKit/UIProcess/API/glib/WebKitWebView.h.in
@@ -71,7 +71,7 @@
 #include <wpe/WebKitColor.h>
 #include <wpe/WebKitRectangle.h>
 #include <wpe/WebKitWebViewBackend.h>
-#if ENABLE(2022_GLIB_API)
+#if ENABLE(WPE_PLATFORM)
 #include <wpe/wpe-platform.h>
 #endif
 #endif
@@ -475,7 +475,7 @@ webkit_web_view_new_with_user_content_ma
 WEBKIT_API WebKitWebViewBackend *
 webkit_web_view_get_backend                          (WebKitWebView             *web_view);
 
-#if ENABLE(2022_GLIB_API)
+#if ENABLE(WPE_PLATFORM)
 WEBKIT_API WPEDisplay *
 webkit_web_view_get_display                          (WebKitWebView             *web_view);
 
Index: webkitgtk/Source/WebKit/UIProcess/API/wpe/WebKitWebViewWPE.cpp
===================================================================
--- webkitgtk.orig/Source/WebKit/UIProcess/API/wpe/WebKitWebViewWPE.cpp
+++ webkitgtk/Source/WebKit/UIProcess/API/wpe/WebKitWebViewWPE.cpp
@@ -295,7 +295,7 @@ guint createShowOptionMenuSignal(WebKitW
         WEBKIT_TYPE_RECTANGLE | G_SIGNAL_TYPE_STATIC_SCOPE);
 }
 
-#if ENABLE(2022_GLIB_API)
+#if ENABLE(WPE_PLATFORM)
 /**
  * webkit_web_view_toggle_inspector:
  * @web_view: a #WebKitWebView
Index: webkitgtk/Source/cmake/OptionsWPE.cmake
===================================================================
--- webkitgtk.orig/Source/cmake/OptionsWPE.cmake
+++ webkitgtk/Source/cmake/OptionsWPE.cmake
@@ -96,6 +96,7 @@ endif ()
 WEBKIT_OPTION_DEFINE(ENABLE_DOCUMENTATION "Whether to generate documentation." PUBLIC ON)
 WEBKIT_OPTION_DEFINE(ENABLE_INTROSPECTION "Whether to enable GObject introspection." PUBLIC ON)
 WEBKIT_OPTION_DEFINE(ENABLE_JOURNALD_LOG "Whether to enable journald logging" PUBLIC ON)
+WEBKIT_OPTION_DEFINE(ENABLE_WPE_PLATFORM "Whether to enable the WPEPlatform library" PUBLIC ${ENABLE_DEVELOPER_MODE})
 WEBKIT_OPTION_DEFINE(ENABLE_WPE_PLATFORM_DRM "Whether to enable support for DRM platform" PUBLIC ON)
 WEBKIT_OPTION_DEFINE(ENABLE_WPE_PLATFORM_HEADLESS "Whether to enable support for headless platform" PUBLIC ON)
 WEBKIT_OPTION_DEFINE(ENABLE_WPE_PLATFORM_WAYLAND "Whether to enable support for Wayland platform" PUBLIC ON)
@@ -113,6 +114,8 @@ WEBKIT_OPTION_DEFINE(USE_EXTERNAL_HOLEPU
 WEBKIT_OPTION_DEFINE(USE_SYSPROF_CAPTURE "Whether to use libsysprof-capture for tracing." PRIVATE ON)
 WEBKIT_OPTION_DEFINE(USE_SYSTEM_SYSPROF_CAPTURE "Whether to use a system-provided libsysprof-capture" PRIVATE ${USE_SYSTEM_SYSPROF_CAPTURE_DEFAULT})
 
+WEBKIT_OPTION_CONFLICT(ENABLE_WPE_PLATFORM ENABLE_WPE_1_1_API)
+
 WEBKIT_OPTION_DEPEND(ENABLE_DOCUMENTATION ENABLE_INTROSPECTION)
 WEBKIT_OPTION_DEPEND(USE_SYSTEM_SYSPROF_CAPTURE USE_SYSPROF_CAPTURE)
 
@@ -181,8 +184,6 @@ else ()
 endif ()
 find_package(GLIB ${GLIB_MINIMUM_VERSION} REQUIRED COMPONENTS gio gio-unix gobject gthread gmodule)
 
-SET_AND_EXPOSE_TO_BUILD(ENABLE_WPE_PLATFORM ${ENABLE_2022_GLIB_API})
-
 if (WPE_API_VERSION VERSION_EQUAL "1.1")
     CALCULATE_LIBRARY_VERSIONS_FROM_LIBTOOL_TRIPLE(WEBKIT 8 4 8)
 else ()
