From: Michael Catanzaro <mcatanzaro@redhat.com>
Subject: Disable SKIA on big-endian machines since it's not supported
Bug: https://bugs.webkit.org/show_bug.cgi?id=274032
Bug-Debian: https://bugs.debian.org/1070865
Origin: https://github.com/WebKit/WebKit/pull/28429
Index: webkitgtk/Source/cmake/OptionsGTK.cmake
===================================================================
--- webkitgtk.orig/Source/cmake/OptionsGTK.cmake
+++ webkitgtk/Source/cmake/OptionsGTK.cmake
@@ -140,7 +140,12 @@ WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_
 WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_WEB_API_STATISTICS PRIVATE ON)
 WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_WEB_CODECS PRIVATE ON)
 WEBKIT_OPTION_DEFAULT_PORT_VALUE(ENABLE_WEB_RTC PRIVATE ${ENABLE_EXPERIMENTAL_FEATURES})
-WEBKIT_OPTION_DEFAULT_PORT_VALUE(USE_SKIA PRIVATE ON)
+
+if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.20" AND ${CMAKE_CXX_BYTE_ORDER} STREQUAL "LITTLE_ENDIAN")
+    WEBKIT_OPTION_DEFAULT_PORT_VALUE(USE_SKIA PRIVATE ON)
+else ()
+    WEBKIT_OPTION_DEFAULT_PORT_VALUE(USE_SKIA PRIVATE OFF)
+endif ()
 
 include(GStreamerDependencies)
 
