From: Alberto Garcia <berto@igalia.com>
Subject: Don't check for SSE2 support on i386
Bug-Debian: https://bugs.debian.org/930935
Forwarded: no
Index: webkitgtk/Source/cmake/WebKitCompilerFlags.cmake
===================================================================
--- webkitgtk.orig/Source/cmake/WebKitCompilerFlags.cmake
+++ webkitgtk/Source/cmake/WebKitCompilerFlags.cmake
@@ -216,15 +216,6 @@ if (COMPILER_IS_GCC_OR_CLANG)
         WEBKIT_PREPEND_GLOBAL_COMPILER_FLAGS(-fmax-errors=20)
     endif ()
 
-    # Force SSE2 fp on x86 builds.
-    if (WTF_CPU_X86 AND NOT CMAKE_CROSSCOMPILING)
-        WEBKIT_PREPEND_GLOBAL_COMPILER_FLAGS(-msse2 -mfpmath=sse)
-        include(DetectSSE2)
-        if (NOT SSE2_SUPPORT_FOUND)
-            message(FATAL_ERROR "SSE2 support is required to compile WebKit")
-        endif ()
-    endif ()
-
     # Makes builds faster. The GCC manual warns about the possibility that the assembler being
     # used may not support input from a pipe, but in practice the toolchains we support all do.
     WEBKIT_PREPEND_GLOBAL_COMPILER_FLAGS(-pipe)
Index: webkitgtk/Source/ThirdParty/ANGLE/src/common/mathutil.h
===================================================================
--- webkitgtk.orig/Source/ThirdParty/ANGLE/src/common/mathutil.h
+++ webkitgtk/Source/ThirdParty/ANGLE/src/common/mathutil.h
@@ -550,7 +550,7 @@ inline R roundToNearest(T input)
 {
     static_assert(std::is_floating_point<T>::value);
     static_assert(std::numeric_limits<R>::is_integer);
-#if defined(__aarch64__) || defined(_M_ARM64)
+#if defined(__aarch64__) || defined(_M_ARM64) || defined(__i386__)
     // On armv8, this expression is compiled to a dedicated round-to-nearest instruction
     return static_cast<R>(std::round(input));
 #else
