From: Alberto Garcia <berto@igalia.com>
Subject: Fix non-unified build
Forwarded: no
Index: webkitgtk/Source/JavaScriptCore/runtime/BufferMemoryHandle.cpp
===================================================================
--- webkitgtk.orig/Source/JavaScriptCore/runtime/BufferMemoryHandle.cpp
+++ webkitgtk/Source/JavaScriptCore/runtime/BufferMemoryHandle.cpp
@@ -35,6 +35,8 @@
 #include <wtf/DataLog.h>
 #include <wtf/Gigacage.h>
 #include <wtf/Lock.h>
+#include <wtf/NeverDestroyed.h>
+#include <wtf/OSAllocator.h>
 #include <wtf/Platform.h>
 #include <wtf/PrintStream.h>
 #include <wtf/SafeStrerror.h>
