From: Alberto Garcia <berto@igalia.com>
Subject: Fix non-unified build
Forwarded: no
Index: webkitgtk/Source/WebCore/css/CSSBackgroundRepeatValue.cpp
===================================================================
--- webkitgtk.orig/Source/WebCore/css/CSSBackgroundRepeatValue.cpp
+++ webkitgtk/Source/WebCore/css/CSSBackgroundRepeatValue.cpp
@@ -26,6 +26,9 @@
 #include "config.h"
 #include "CSSBackgroundRepeatValue.h"
 
+#include "CSSValueKeywords.h"
+#include <wtf/text/WTFString.h>
+
 namespace WebCore {
 
 CSSBackgroundRepeatValue::CSSBackgroundRepeatValue(CSSValueID xValue, CSSValueID yValue)
Index: webkitgtk/Source/WebCore/css/CSSContentDistributionValue.cpp
===================================================================
--- webkitgtk.orig/Source/WebCore/css/CSSContentDistributionValue.cpp
+++ webkitgtk/Source/WebCore/css/CSSContentDistributionValue.cpp
@@ -26,6 +26,9 @@
 #include "config.h"
 #include "CSSContentDistributionValue.h"
 
+#include "CSSValueKeywords.h"
+#include <wtf/text/WTFString.h>
+
 namespace WebCore {
 
 CSSContentDistributionValue::CSSContentDistributionValue(CSSValueID distribution, CSSValueID position, CSSValueID overflow)
Index: webkitgtk/Source/WebCore/css/parser/CSSParserContext.cpp
===================================================================
--- webkitgtk.orig/Source/WebCore/css/parser/CSSParserContext.cpp
+++ webkitgtk/Source/WebCore/css/parser/CSSParserContext.cpp
@@ -27,6 +27,7 @@
 #include "CSSParserContext.h"
 
 #include "CSSPropertyNames.h"
+#include "CSSValuePool.h"
 #include "Document.h"
 #include "DocumentLoader.h"
 #include "Page.h"
Index: webkitgtk/Source/WebCore/css/CSSContentDistributionValue.h
===================================================================
--- webkitgtk.orig/Source/WebCore/css/CSSContentDistributionValue.h
+++ webkitgtk/Source/WebCore/css/CSSContentDistributionValue.h
@@ -25,7 +25,7 @@
 
 #pragma once
 
-#include "CSSValue.h"
+#include "CSSBackgroundRepeatValue.h"
 
 namespace WebCore {
 
Index: webkitgtk/Source/WebCore/html/HTMLMetaElement.h
===================================================================
--- webkitgtk.orig/Source/WebCore/html/HTMLMetaElement.h
+++ webkitgtk/Source/WebCore/html/HTMLMetaElement.h
@@ -22,13 +22,12 @@
 
 #pragma once
 
+#include "Color.h"
 #include "HTMLElement.h"
 #include "MediaQuery.h"
 
 namespace WebCore {
 
-class Color;
-
 class HTMLMetaElement final : public HTMLElement {
     WTF_MAKE_ISO_ALLOCATED(HTMLMetaElement);
 public:
