From: Alberto Garcia <berto@igalia.com>
Subject: Define PATH_MAX if it's not already defined
Forwarded: no
Index: webkitgtk/Source/WebCore/platform/text/hyphen/HyphenationLibHyphen.cpp
===================================================================
--- webkitgtk.orig/Source/WebCore/platform/text/hyphen/HyphenationLibHyphen.cpp
+++ webkitgtk/Source/WebCore/platform/text/hyphen/HyphenationLibHyphen.cpp
@@ -42,6 +42,10 @@
 #include <wtf/text/CString.h>
 #include <wtf/text/StringView.h>
 
+#if !defined(PATH_MAX)
+#define PATH_MAX 4096
+#endif
+
 namespace WebCore {
 
 static const char* const gDictionaryDirectories[] = {
Index: webkitgtk/Source/JavaScriptCore/runtime/ConfigFile.h
===================================================================
--- webkitgtk.orig/Source/JavaScriptCore/runtime/ConfigFile.h
+++ webkitgtk/Source/JavaScriptCore/runtime/ConfigFile.h
@@ -42,6 +42,8 @@ private:
 
 #if OS(WINDOWS)
     static const size_t s_maxPathLength = 260; // Windows value for "MAX_PATH"
+#elif OS(HURD)
+    static const size_t s_maxPathLength = 4096;
 #else
     static const size_t s_maxPathLength = PATH_MAX;
 #endif
