From: Alexander Kanavin <alex@linutronix.de>
Subject: Ensure reproducibility of __TIMESTAMP__
Bug: https://bugs.webkit.org/show_bug.cgi?id=252308
Origin: https://github.com/WebKit/WebKit/pull/15293
Index: webkitgtk/Source/JavaScriptCore/CMakeLists.txt
===================================================================
--- webkitgtk.orig/Source/JavaScriptCore/CMakeLists.txt
+++ webkitgtk/Source/JavaScriptCore/CMakeLists.txt
@@ -159,6 +159,7 @@ add_custom_command(
     OUTPUT ${JavaScriptCore_DERIVED_SOURCES_DIR}/JSCBytecodeCacheVersion.cpp
     MAIN_DEPENDENCY ${JAVASCRIPTCORE_DIR}/runtime/JSCBytecodeCacheVersion.cpp.in
     COMMAND ${PERL_EXECUTABLE} -pe s/CACHED_TYPES_CKSUM/__TIMESTAMP__/ ${JAVASCRIPTCORE_DIR}/runtime/JSCBytecodeCacheVersion.cpp.in > ${JavaScriptCore_DERIVED_SOURCES_DIR}/JSCBytecodeCacheVersion.cpp
+    COMMAND touch -r ${JAVASCRIPTCORE_DIR}/runtime/JSCBytecodeCacheVersion.cpp.in ${JavaScriptCore_DERIVED_SOURCES_DIR}/JSCBytecodeCacheVersion.cpp
             VERBATIM
 )
 
