From: Žan Doberšek <zdobersek@igalia.com>
Subject: Fix RISCV64 build
Origin: https://github.com/WebKit/WebKit/commit/bb7a34a9631ccf0d34eedc78ac0de137916ef390
Index: webkitgtk/Source/JavaScriptCore/assembler/MacroAssemblerRISCV64.h
===================================================================
--- webkitgtk.orig/Source/JavaScriptCore/assembler/MacroAssemblerRISCV64.h
+++ webkitgtk/Source/JavaScriptCore/assembler/MacroAssemblerRISCV64.h
@@ -1800,6 +1800,10 @@ public:
         m_assembler.fmvInsn<RISCV64Assembler::FMVType::D, RISCV64Assembler::FMVType::X>(dest, src);
     }
 
+    static bool supportsCountPopulation() { return false; }
+    MACRO_ASSEMBLER_RISCV64_TEMPLATED_NOOP_METHOD(countPopulation32);
+    MACRO_ASSEMBLER_RISCV64_TEMPLATED_NOOP_METHOD(countPopulation64);
+
     // The RISC-V V vector extension is not yet standardized
     MACRO_ASSEMBLER_RISCV64_TEMPLATED_NOOP_METHOD(moveVector);
     MACRO_ASSEMBLER_RISCV64_TEMPLATED_NOOP_METHOD(loadVector);
