From: Tomas Popela <tpopela@redhat.com>
Subject: Fix FTBFS because of duplicate symbols in Plugin{Package,View}None
Bug: https://bugs.webkit.org/show_bug.cgi?id=140241
Index: webkitgtk/Source/WebCore/GNUmakefile.list.am
===================================================================
--- webkitgtk.orig/Source/WebCore/GNUmakefile.list.am
+++ webkitgtk/Source/WebCore/GNUmakefile.list.am
@@ -6254,9 +6254,13 @@ endif # END USE_GLX
 endif # END TARGET_X11
 
 if TARGET_WAYLAND
+if !TARGET_X11
+if ENABLE_WEBKIT2
 webcore_sources += \
 	Source/WebCore/plugins/PluginPackageNone.cpp \
 	Source/WebCore/plugins/PluginViewNone.cpp
+endif # END ENABLE_WEBKIT2
+endif # END !TARGET_X11
 endif # END TARGET_WAYLAND
 
 if TARGET_X11_OR_WAYLAND
@@ -6305,12 +6309,16 @@ webcoregtk_sources += \
 endif # END TARGET_WIN32
 
 if TARGET_QUARTZ
+if !TARGET_X11
+if ENABLE_WEBKIT2
 webcore_sources += \
 	Source/WebCore/plugins/PluginPackageNone.cpp \
 	Source/WebCore/plugins/PluginViewNone.cpp
 platformgtk_sources += \
 	Source/WebCore/platform/cairo/WidgetBackingStoreCairo.h \
 	Source/WebCore/platform/cairo/WidgetBackingStoreCairo.cpp
+endif # END ENABLE_WEBKIT2
+endif # END !TARGET_X11
 endif # END TARGET_QUARTZ
 
 if USE_GLES2
