From: Adrian Perez de Castro <aperez@igalia.com>
Subject: Fix WPEWebKit API 2.0 library soname
Origin: https://github.com/WebKit/WebKit/commit/173c1e5f57615df8f4c523b308f9c419e8f7c9d9
Index: webkitgtk/Source/cmake/OptionsWPE.cmake
===================================================================
--- webkitgtk.orig/Source/cmake/OptionsWPE.cmake
+++ webkitgtk/Source/cmake/OptionsWPE.cmake
@@ -182,7 +182,7 @@ EXPOSE_STRING_VARIABLE_TO_BUILD(WPE_API_
 if (WPE_API_VERSION VERSION_EQUAL "1.1")
     CALCULATE_LIBRARY_VERSIONS_FROM_LIBTOOL_TRIPLE(WEBKIT 9 5 9)
 else ()
-    CALCULATE_LIBRARY_VERSIONS_FROM_LIBTOOL_TRIPLE(WEBKIT 6 5 4)
+    CALCULATE_LIBRARY_VERSIONS_FROM_LIBTOOL_TRIPLE(WEBKIT 6 5 5)
 endif ()
 
 if (ENABLE_WPE_PLATFORM)
