From: Alberto Garcia <berto@igalia.com>
Subject: Increase the page size so it works fine in Loongson
Bug: https://bugs.webkit.org/show_bug.cgi?id=221198
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=977779
Index: webkitgtk/Source/WTF/wtf/PageBlock.h
===================================================================
--- webkitgtk.orig/Source/WTF/wtf/PageBlock.h
+++ webkitgtk/Source/WTF/wtf/PageBlock.h
@@ -51,7 +51,9 @@ namespace WTF {
 constexpr size_t CeilingOnPageSize = 16 * KB;
 #elif USE(64KB_PAGE_BLOCK) || CPU(PPC) || CPU(PPC64) || CPU(PPC64LE) || CPU(UNKNOWN)
 constexpr size_t CeilingOnPageSize = 64 * KB;
-#elif OS(WINDOWS) || CPU(MIPS) || CPU(MIPS64) || CPU(X86) || CPU(X86_64) || CPU(ARM) || CPU(ARM64)
+#elif CPU(MIPS) || CPU(MIPS64)
+constexpr size_t CeilingOnPageSize = 16 * KB;
+#elif OS(WINDOWS) || CPU(X86) || CPU(X86_64) || CPU(ARM) || CPU(ARM64)
 constexpr size_t CeilingOnPageSize = 4 * KB;
 #else
 #error Must set CeilingOnPageSize in PageBlock.h when adding a new CPU architecture!
