From: Diego Pino Garcia <dpino@igalia.com>
Subject: Fix build failure with older versions of GStreamer
Origin: https://commits.webkit.org/256481@main
Index: webkitgtk/Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp
===================================================================
--- webkitgtk.orig/Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp
+++ webkitgtk/Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp
@@ -856,9 +856,11 @@ void fillVideoInfoColorimetryFromColorSp
         case PlatformVideoTransferCharacteristics::Bt709:
             GST_VIDEO_INFO_COLORIMETRY(info).transfer = GST_VIDEO_TRANSFER_BT709;
             break;
+#if GST_CHECK_VERSION(1, 18, 0)
         case PlatformVideoTransferCharacteristics::Smpte170m:
             GST_VIDEO_INFO_COLORIMETRY(info).transfer = GST_VIDEO_TRANSFER_BT601;
             break;
+#endif
         default:
             break;
         }
