From: Philippe Normand <pnormand@igalia.com>
Subject: Fix playback of redirected streams
Bug: https://bugs.webkit.org/show_bug.cgi?id=197410
Bug-Debian: https://bugs.debian.org/928044
Origin: https://trac.webkit.org/changeset/245054/webkit
Index: webkitgtk/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
===================================================================
--- webkitgtk.orig/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
+++ webkitgtk/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
@@ -1350,9 +1350,8 @@ void MediaPlayerPrivateGStreamer::handle
         }
 #endif
         else if (gst_structure_has_name(structure, "http-headers")) {
-            const char* redirectionUri = gst_structure_get_string(structure, "redirection-uri");
-            const char* uri = redirectionUri ? redirectionUri : gst_structure_get_string(structure, "uri");
-            if (uri) {
+            GST_DEBUG_OBJECT(pipeline(), "Processing HTTP headers: %" GST_PTR_FORMAT, structure);
+            if (const char* uri = gst_structure_get_string(structure, "uri")) {
                 URL url(URL(), uri);
 
                 m_origins.add(SecurityOrigin::create(url));
