From: Kimmo Kinnunen <kkinnunen@apple.com>
Subject: Fix build in 32-bit systems due to wrong divisor in round operation
Bug: https://bugs.webkit.org/show_bug.cgi?id=222480
Index: webkitgtk/Source/WebKit/Platform/IPC/StreamConnectionBuffer.h
===================================================================
--- webkitgtk.orig/Source/WebKit/Platform/IPC/StreamConnectionBuffer.h
+++ webkitgtk/Source/WebKit/Platform/IPC/StreamConnectionBuffer.h
@@ -114,7 +114,7 @@ private:
         alignas(sizeof(uint64_t[2])) Atomic<size_t> senderOffset;
     };
     Header& header() const { return *reinterpret_cast<Header*>(m_sharedMemory->data()); }
-    static constexpr size_t headerSize() { return roundUpToMultipleOf<sizeof(std::max_align_t)>(sizeof(Header)); }
+    static constexpr size_t headerSize() { return roundUpToMultipleOf<alignof(std::max_align_t)>(sizeof(Header)); }
 
     size_t m_dataSize { 0 };
     Ref<WebKit::SharedMemory> m_sharedMemory;
