From: Alberto Garcia <berto@igalia.com>
Subject: Fix non-unified build
Bug: https://bugs.webkit.org/show_bug.cgi?id=246816
Index: webkitgtk/Source/JavaScriptCore/interpreter/StackVisitor.cpp
===================================================================
--- webkitgtk.orig/Source/JavaScriptCore/interpreter/StackVisitor.cpp
+++ webkitgtk/Source/JavaScriptCore/interpreter/StackVisitor.cpp
@@ -28,6 +28,7 @@
 
 #include "ClonedArguments.h"
 #include "DebuggerPrimitives.h"
+#include "ExecutableBaseInlines.h"
 #include "InlineCallFrame.h"
 #include "JSCInlines.h"
 #include "RegisterAtOffsetList.h"
