From: Alberto Garcia <berto@igalia.com>
Subject: Fix crash in the Flash plugin
Bug: https://bugs.webkit.org/show_bug.cgi?id=137849
Bug-Debian: http://bugs.debian.org/768929
Index: webkitgtk/Source/WebKit2/WebProcess/Plugins/Netscape/x11/NetscapePluginX11.cpp
===================================================================
--- webkitgtk.orig/Source/WebKit2/WebProcess/Plugins/Netscape/x11/NetscapePluginX11.cpp
+++ webkitgtk/Source/WebKit2/WebProcess/Plugins/Netscape/x11/NetscapePluginX11.cpp
@@ -201,7 +201,9 @@ void NetscapePlugin::platformPreInitiali
 bool NetscapePlugin::platformPostInitialize()
 {
     uint64_t windowID = 0;
-    bool needsXEmbed = false;
+    // NPPVpluginNeedsXEmbed is a boolean value, but at least the
+    // Flash player plugin is using an 'int' instead.
+    int needsXEmbed = 0;
     if (m_isWindowed) {
         NPP_GetValue(NPPVpluginNeedsXEmbed, &needsXEmbed);
         if (needsXEmbed) {
