From: Ting-Wei Lan <lantw44@gmail.com>
Subject: gtkdoc-fixxref needs --module argument to work
Bug: https://bugs.webkit.org/show_bug.cgi?id=172415
Origin: http://trac.webkit.org/changeset/217190
Index: webkitgtk/Tools/gtk/gtkdoc.py
===================================================================
--- webkitgtk.orig/Tools/gtk/gtkdoc.py
+++ webkitgtk/Tools/gtk/gtkdoc.py
@@ -366,6 +366,7 @@ class GTKDoc(object):
 
     def _run_gtkdoc_fixxref(self):
         args = ['gtkdoc-fixxref',
+                '--module=%s' % self.module_name,
                 '--module-dir=html',
                 '--html-dir=html']
         args.extend(['--extra-dir=%s' % extra_dir for extra_dir in self.cross_reference_deps])
