From: Carlos Garcia Campos <cgarcia@igalia.com>
Subject: Fix build with wpebackend-fdo 1.8.1
Origin: https://trac.webkit.org/changeset/273376
Index: webkitgtk/Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp
===================================================================
--- webkitgtk.orig/Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp
+++ webkitgtk/Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp
@@ -42,6 +42,7 @@
 #include <gtk/gtk.h>
 
 #if PLATFORM(WAYLAND) && USE(WPE_RENDERER)
+#include <wpe/wpe.h>
 #include <wpe/fdo.h>
 #endif
 #endif
Index: webkitgtk/Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreWayland.cpp
===================================================================
--- webkitgtk.orig/Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreWayland.cpp
+++ webkitgtk/Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreWayland.cpp
@@ -50,6 +50,7 @@
 #endif
 
 #if USE(WPE_RENDERER)
+#include <wpe/wpe.h>
 #include <wpe/fdo-egl.h>
 #if WPE_FDO_CHECK_VERSION(1, 7, 0)
 #include <wayland-server.h>
