From: Carlos Alberto Lopez Perez <clopez@igalia.com>
Subject: Don't use a SSL3.0 record version in client hello
Origin: http://trac.webkit.org/changeset/176253
Index: webkitgtk/Source/WebKit2/gtk/MainGtk.cpp
===================================================================
--- webkitgtk.orig/Source/WebKit2/gtk/MainGtk.cpp
+++ webkitgtk/Source/WebKit2/gtk/MainGtk.cpp
@@ -38,7 +38,7 @@ int main(int argc, char** argv)
     // overwrite this priority string if it's already set by the user.
     // Keep this in sync with WebProcessMain.cpp.
     // https://bugzilla.gnome.org/show_bug.cgi?id=738633
-    setenv("G_TLS_GNUTLS_PRIORITY", "NORMAL:%COMPAT:!VERS-SSL3.0", 0);
+    setenv("G_TLS_GNUTLS_PRIORITY", "NORMAL:%COMPAT:%LATEST_RECORD_VERSION:!VERS-SSL3.0", 0);
 
     return WebKit::WebProcessMainGtk(argc, argv);
 }
Index: webkitgtk/Source/WebKit2/unix/NetworkMainUnix.cpp
===================================================================
--- webkitgtk.orig/Source/WebKit2/unix/NetworkMainUnix.cpp
+++ webkitgtk/Source/WebKit2/unix/NetworkMainUnix.cpp
@@ -42,7 +42,7 @@ int main(int argc, char** argv)
     // overwrite this priority string if it's already set by the user.
     // Keep this in sync with WebProcessMain.cpp.
     // https://bugzilla.gnome.org/show_bug.cgi?id=738633
-    setenv("G_TLS_GNUTLS_PRIORITY", "NORMAL:%COMPAT:!VERS-SSL3.0", 0);
+    setenv("G_TLS_GNUTLS_PRIORITY", "NORMAL:%COMPAT:%LATEST_RECORD_VERSION:!VERS-SSL3.0", 0);
 
     return WebKit::NetworkProcessMain(argc, argv);
 }
