From: Csaba Osztrogonác <ossy@webkit.org>
Subject: Check for NULL pointer in ApplyStyleCommand
Origin: http://trac.webkit.org/changeset/176182
Index: webkitgtk/Source/WebCore/editing/ApplyStyleCommand.cpp
===================================================================
--- webkitgtk.orig/Source/WebCore/editing/ApplyStyleCommand.cpp
+++ webkitgtk/Source/WebCore/editing/ApplyStyleCommand.cpp
@@ -267,6 +267,9 @@ void ApplyStyleCommand::applyBlockStyle(
 #else
     Node* scope = highestEditableRoot(visibleStart.deepEquivalent());
 #endif
+    if (!scope)
+        return;
+
     RefPtr<Range> startRange = Range::create(document(), firstPositionInNode(scope), visibleStart.deepEquivalent().parentAnchoredEquivalent());
     RefPtr<Range> endRange = Range::create(document(), firstPositionInNode(scope), visibleEnd.deepEquivalent().parentAnchoredEquivalent());
     int startIndex = TextIterator::rangeLength(startRange.get(), true);
