From: Zalan Bujtas <zalan@apple.com>
Subject: Check for NULL pointer in SearchInputType.
Origin: http://trac.webkit.org/changeset/175912
Index: webkitgtk/Source/WebCore/html/SearchInputType.cpp
===================================================================
--- webkitgtk.orig/Source/WebCore/html/SearchInputType.cpp
+++ webkitgtk/Source/WebCore/html/SearchInputType.cpp
@@ -178,7 +178,7 @@ bool SearchInputType::searchEventsShould
 
 void SearchInputType::didSetValueByUserEdit(ValueChangeState state)
 {
-    if (m_cancelButton)
+    if (m_cancelButton && element().renderer())
         toRenderSearchField(element().renderer())->updateCancelButtonVisibility();
 
     // If the incremental attribute is set, then dispatch the search event
