From: Tomas Popela <tpopela@redhat.com>
Subject: Use ternary operator instead of std::max()
Bug: https://bugs.webkit.org/show_bug.cgi?id=189078
Index: webkitgtk/Source/JavaScriptCore/heap/RegisterState.h
===================================================================
--- webkitgtk.orig/Source/JavaScriptCore/heap/RegisterState.h
+++ webkitgtk/Source/JavaScriptCore/heap/RegisterState.h
@@ -166,7 +166,7 @@ using RegisterState = jmp_buf;
 #pragma warning(disable: 4611)
 #endif
 #define ALLOCATE_AND_GET_REGISTER_STATE(registers) \
-    alignas(std::max(alignof(void*), alignof(RegisterState))) RegisterState registers; \
+    alignas(alignof(void*) > alignof(RegisterState) ? alignof(void*) : alignof(RegisterState)) RegisterState registers; \
     setjmp(registers)
 
 #if COMPILER(MSVC)
