From: Miyoung Shin <myid.shin@samsung.com>
Subject: Fix crash during dispatching touchEvent created by JS
Bug-Debian: https://bugs.debian.org/761492
Bug: https://bugs.webkit.org/show_bug.cgi?id=138211
Index: webkitgtk/Source/WebCore/dom/EventDispatcher.cpp
===================================================================
--- webkitgtk.orig/Source/WebCore/dom/EventDispatcher.cpp
+++ webkitgtk/Source/WebCore/dom/EventDispatcher.cpp
@@ -91,7 +91,7 @@ public:
     EventContext& contextAt(size_t i) { return *m_path[i]; }
 
 #if ENABLE(TOUCH_EVENTS)
-    void updateTouchLists(const TouchEvent&);
+    bool updateTouchLists(const TouchEvent&);
 #endif
     void setRelatedTarget(EventTarget&);
 
@@ -312,8 +312,10 @@ bool EventDispatcher::dispatchEvent(Node
     if (EventTarget* relatedTarget = event->relatedTarget())
         eventPath.setRelatedTarget(*relatedTarget);
 #if ENABLE(TOUCH_EVENTS) && !PLATFORM(IOS)
-    if (event->isTouchEvent())
-        eventPath.updateTouchLists(*toTouchEvent(event.get()));
+    if (event->isTouchEvent()) {
+        if (!eventPath.updateTouchLists(*toTouchEvent(event.get())))
+            return true;
+    }
 #endif
 
     ChildNodesLazySnapshot::takeChildNodesLazySnapshot();
@@ -432,8 +434,11 @@ static void addRelatedNodeResolversForTo
         touchTargetResolvers.append(EventRelatedNodeResolver(*touchList->item(i), type));
 }
 
-void EventPath::updateTouchLists(const TouchEvent& touchEvent)
+bool EventPath::updateTouchLists(const TouchEvent& touchEvent)
 {
+    if (!touchEvent.touches() || !touchEvent.targetTouches() || !touchEvent.changedTouches())
+        return false;
+
     Vector<EventRelatedNodeResolver, 16> touchTargetResolvers;
     const size_t touchNodeCount = touchEvent.touches()->length() + touchEvent.targetTouches()->length() + touchEvent.changedTouches()->length();
     touchTargetResolvers.reserveInitialCapacity(touchNodeCount);
@@ -454,6 +459,7 @@ void EventPath::updateTouchLists(const T
             context.touchList(currentResolver.touchListType())->append(currentResolver.touch()->cloneWithNewTarget(nodeInCurrentTreeScope));
         }
     }
+    return true;
 }
 #endif
 
