From: Alberto Garcia <berto@igalia.com>
Subject: Default to OpenGL ES in arm
Bug: https://bugs.webkit.org/show_bug.cgi?id=214828
Origin: https://trac.webkit.org/changeset/264923/webkit
Index: webkitgtk/Source/cmake/OptionsGTK.cmake
===================================================================
--- webkitgtk.orig/Source/cmake/OptionsGTK.cmake
+++ webkitgtk/Source/cmake/OptionsGTK.cmake
@@ -69,7 +69,7 @@ endif ()
 # or OPENGLES2 (ENABLE_GLES2=ON) for building the feature.
 set(ENABLE_GLES2_DEFAULT OFF)
 
-if (NOT OPENGL_FOUND AND OPENGLES2_FOUND)
+if (OPENGLES2_FOUND AND (NOT OPENGL_FOUND OR WTF_CPU_ARM OR WTF_CPU_ARM64))
     set(ENABLE_GLES2_DEFAULT ON)
 endif ()
 
