From: Carlos Garcia Campos <cgarcia@igalia.com>
Subject: Fix name of the WebKitWebDriver executable
Bug: https://bugs.webkit.org/show_bug.cgi?id=198480
Origin: https://trac.webkit.org/changeset/246035/webkit
Index: webkitgtk/Source/WebDriver/PlatformGTK.cmake
===================================================================
--- webkitgtk.orig/Source/WebDriver/PlatformGTK.cmake
+++ webkitgtk/Source/WebDriver/PlatformGTK.cmake
@@ -1,4 +1,4 @@
-set(WebDriver_Process_OUTPUT_NAME WebKitWebDriver)
+set(WebDriver_OUTPUT_NAME WebKitWebDriver)
 
 add_definitions(-DLIBEXECDIR="${CMAKE_INSTALL_FULL_LIBEXECDIR}")
 
